src/shapes/audio_token.class.js

```diff
@@ -0,0 +1,202 @@
+  /**
+   * Audio_token class
+   * @class fabric.Audio_token
+   * @extends fabric.Object
+   * @see {@link fabric.Audio_token#initialize} for constructor definition
+   */
+  fabric.Audio_token = fabric.util.createClass(fabric.Object, /** @lends fabric.Audio_token.prototype */ {
+
+    /**
+     * Type of an object
+     * @type String
+     * @default
+     */
+    type: 'audio_token',
+
+    /**
+     * unique key - used for now just to confirm play/pause for specific tokens
+     * @since 2.0.0
+     * @type String
+     * @default
+     */
+    cacheKey: '',
+
+    /**
+     * Indicates whether audio is in 'play' mode
+     * @type Boolean
+     * @default
+     */
+    isPlaying: false,
+
+    /**
+     * Indicates whether audio is in 'paused' mode
+     * @type Boolean
+     * @default
+     */
+    isPaused: false,
+
+    /**
+     * Indicates where audio playback is currently (in ms) //not used yet
+     * @type number
+     * @default
+     */
+    playbackMS: 0,
+
+    /**
+     * Audio URL? WIP - not sure how this will be used
+     * To be determined by: NEWXP-1949, likely passed in as a parameter on creation
+     * @type string
+     * @default
+     */
+    audioUrl: '',
+
+    hasBorders: false,
+
+    /**
+     * extra space taken up on the left by delete controls (used for collision detection)
+     * @type Number
+     * @default
+     */
+    leftMargin: 44,
+
+    /**
+     * Definable images for idle state, selected state, and play/pause controls
+     */
+    idleImage: null,
+    selectedImage: null,
+    playControlImage: null,
+    pauseControlImage: null,
+
+    idleImageSrc: undefined,
+    selectedImageSrc: undefined,
+    playControlImageSrc: undefined,
+    pauseControlImageSrc: undefined,
+
+    /**
+     * List of properties to consider when checking if
+     * state of an object is changed ({@link fabric.Object#hasStateChanged})
+     * @type Array
+     */
+    stateProperties: fabric.Object.prototype.stateProperties.concat('isPlaying', 'isPaused'),
+
+    /**
+     * Constructor
+     * @param {Object} [options] Options object
+     * @return {fabric.Audio_token} thisArg
+     */
+    initialize: function(audioURL, options) {
+      options || (options = { });
+      this.filters = [];
+      this.audioURL = audioURL;
+      this.cacheKey = 'audio_token' + fabric.Object.__uid++;
+      this.callSuper('initialize', options);
+      this.initBehavior();
+      this.initImages();
+    },
+
+    /**
+     * Draws a background for the object big as its untransformed dimensions
+     * @private
+     * @param {CanvasRenderingContext2D} ctx Context to render on
+     */
+    _renderBackground: function() {
+      return;
+    },
+
+    _isSelected: function() {
+      return this.canvas && this.canvas._activeObject === this;
+    },
+
+    initImages: function() {
+      if (this.idleImageSrc) {
+        //TODO: will this create multiple images if we have multiple instances? Or is this a property of the prototype?
```

Comment:
Need some discussion here. Do we want to be able to pass in multiple images (I can make up some cases where we'd want different colored audio tokens to make it easier for students to distinguish multiple ones.)

Reply:
Well, as it stands we have the idleImageSrc and that can be overwritten by amp-worksheet passing in a different base64 png to fabric.Audio_Token.prototype.idleImageSrc  - but if we WANTED each audio token to potentially have different images then we couldn't do that. So my question is 'Do we want to allow that'. (I think the answer is no...I refactored this code several times to keep the src info out of the save files and to keep from making multiple document.createElement('img') for EACH audio token). I guess I was still on the fence when I added that PR comment!